lib/matplotlib/axis.py

```diff
@@ -826,15 +826,24 @@ def set_tick_params(self, which='major', reset=False, **kw):
         For documentation of keyword arguments, see
         :meth:`matplotlib.axes.Axes.tick_params`.
         """
+        blacklist = ['tick1On', 'tick2On', 'label1On', 'label2On', 'gridOn']
         dicts = []
+        blacklist_dicts = []
         if which == 'major' or which == 'both':
             dicts.append(self._major_tick_kw)
         if which == 'minor' or which == 'both':
             dicts.append(self._minor_tick_kw)
         kwtrans = self._translate_tick_kw(kw)
+
         for d in dicts:
             if reset:
                 d.clear()
+            blacklist_dicts.append(
```

Comment:
I find this pretty opaque, and don’t really understand why these kwargs are different than the other kwargs (other than that they control visibility).  This could really use a comment explaining what it does and why.  As it stands, I’m not clear at all that this is the proper solution, and maybe there is something more fundamental that is at fault.